Tests/SentryTests/Profiling/SentryProfilerSwiftTests.swift

```diff
@@ -0,0 +1,354 @@
+        fixture.options.tracesSampleRate = 1.0
+        let expectedEnvironment = "test-environment"
+        fixture.options.environment = expectedEnvironment
+        performTest(transactionEnvironment: expectedEnvironment)
+    }
+
+    func testProfilingDataContainsEnvironmentSetFromConfigureScope() {
+        fixture.options.profilesSampleRate = 1.0
+        fixture.options.tracesSampleRate = 1.0
+        let expectedEnvironment = "test-environment"
+        fixture.hub.configureScope { scope in
+            scope.setEnvironment(expectedEnvironment)
+        }
+        performTest(transactionEnvironment: expectedEnvironment)
+    }
+
+    func testStartTransaction_NotSamplingProfileUsingEnableProfiling() {
+        assertProfilesSampler(expectedDecision: .no) { options in
+            options.enableProfiling_DEPRECATED_TEST_ONLY = false
+        }
+    }
+
+    func testStartTransaction_SamplingProfileUsingEnableProfiling() {
+        assertProfilesSampler(expectedDecision: .yes) { options in
+            options.enableProfiling_DEPRECATED_TEST_ONLY = true
+        }
+    }
+
+    func testStartTransaction_NotSamplingProfileUsingSampleRate() {
+        assertProfilesSampler(expectedDecision: .no) { options in
+            options.profilesSampleRate = 0.49
+        }
+    }
+
+    func testStartTransaction_SamplingProfileUsingSampleRate() {
+        assertProfilesSampler(expectedDecision: .yes) { options in
+            options.profilesSampleRate = 0.5
+        }
+    }
+
+    func testStartTransaction_SamplingProfileUsingProfilesSampler() {
+        assertProfilesSampler(expectedDecision: .yes) { options in
+            options.profilesSampler = { _ in return 0.51 }
+        }
+    }
+
+    func testStartTransaction_WhenProfilesSampleRateAndProfilesSamplerNil() {
+        assertProfilesSampler(expectedDecision: .no) { options in
+            options.profilesSampleRate = nil
+            options.profilesSampler = { _ in return nil }
+        }
+    }
+
+    func testStartTransaction_WhenProfilesSamplerOutOfRange_TooBig() {
+        assertProfilesSampler(expectedDecision: .no) { options in
+            options.profilesSampler = { _ in return 1.01 }
+        }
+    }
+
+    func testStartTransaction_WhenProfilesSamplersOutOfRange_TooSmall() {
+        assertProfilesSampler(expectedDecision: .no) { options in
+            options.profilesSampler = { _ in return -0.01 }
+        }
+    }
+}
+
+private extension SentryProfilerSwiftTests {
+    /// Keep a thread busy over a long enough period of time (long enough for 3 samples) for the sampler to pick it up.
+    func forceProfilerSample() {
+        let queue = DispatchQueue(label: "SentryProfilerSwiftTests")
+        let group = DispatchGroup()
+
+        group.enter()
+
+        // Some busy work to try and get it to show up in the profile.
+        let str = "a"
+        var concatStr = ""
+        for _ in 0..<100_000 {
+            concatStr = concatStr.appending(str)
+        }
```

Comment:
I noticed that this part takes very long (a couple of seconds) to execute and will slow down our test suite. Can we speed this up a bit or work our way around it?

Reply:
This was already present in the profiling test logic, it is only showing up as an addition here because I moved it out of SentryHubTests into a more appropriate source file. Let's leave this kind of thing for a separate PR.